Services/Language/classes/class.ilObjLanguageDBAccess.php

```diff
@@ -134,13 +134,29 @@ public function replaceLangModules(array $lang_array): void
                 "(%s,%s,%s),",
                 $this->ilDB->quote($module, "text"),
                 $this->ilDB->quote($this->key, "text"),
-                $this->ilDB->quote(serialize($lang_arr), "clob"),
+                $this->ilDB->quote(serialize($lang_arr), "clob")
             );
         }
-        $this->ilDB->manipulate(sprintf(
-            "DELETE FROM lng_modules WHERE lang_key = %s",
-            $this->ilDB->quote($this->key, "text"),
-        ));
+        
+        $delete_query = sprintf("DELETE FROM lng_modules WHERE lang_key = %s",
+            $this->ilDB->quote($this->key, "text")
+        );
+        if ($this->scope === "local") {
+            // delete only modules containing local changes
+            // see mantis #36972
+            $modules_to_delete = "module = '";
+            $count_modules = count($lang_array);
+            foreach ($lang_array as $module => $lang_arr) {
+                $modules_to_delete .= $module . "'";
+                $count_modules -= 1;
+                if ($count_modules) {
+                    $modules_to_delete .= " OR module = '";
+                }
+            }
+            $delete_query .= " AND (" . $modules_to_delete . ")";
```

Comment:
Please use `ilDBInterface::quote` and don't write values unescaped into the query.